### PR TITLE
[processing] add remaining functionalify from the interpolation plugin

### PR DESCRIPTION
**Pull request by alexbruy**
Also drop Interpolation plugin

### CODE REVIEW

**Comment by nyalldawson** [comment]
@alexbruy you're on fire... This is great work!

**Comment by nirvn** [comment]
Nice!

**Comment by alexbruy** [comment]
@m-kuhn may I ask you to look at my changes to test suite? I added support for custom parameter, used in interpolation algoroithms

File: python/plugins/processing/algs/qgis/IdwInterpolation.py [hunk 1]
```diff
@@ -29,28 +29,28 @@
...
 from processing.core.GeoAlgorithm import GeoAlgorithm
 from processing.core.GeoAlgorithmExecutionException import GeoAlgorithmExecutionException
-from processing.core.parameters import ParameterVector
-from processing.core.parameters import ParameterSelection
-from processing.core.parameters import ParameterNumber
-from processing.core.parameters import ParameterExtent
+from processing.core.parameters import (Parameter,
+                                        ParameterNumber,
+                                        ParameterExtent,
+                                        _splitParameterOptions
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/IdwInterpolation.py]: Is this a public and stable function?
I.e. can the `_`-prefix be removed?

File: python/plugins/processing/algs/qgis/TinInterpolation.py [hunk 2]
```diff
@@ -29,28 +29,31 @@
...
+                                        ParameterExtent,
+                                        ParameterSelection,
+                                        _splitParameterOptions
+                                       )
+from processing.core.outputs import (OutputRaster,
+                                     OutputVector
+                                    )
 from processing.tools import dataobjects
 
 pluginPath = os.path.split(os.path.split(os.path.dirname(__file__))[0])[0]
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/TinInterpolation.py]: Does this double-split look look strange to you as well?
(I realize that it wasn't changed in this pull request)

File: python/plugins/processing/algs/qgis/IdwInterpolation.py [hunk 3]
```diff
@@ -63,20 +63,61 @@ def getIcon(self):
...
+                    return True
+
+                if value == '':
+                   if not self.optional:
+                       return False
+
+                if isinstance(value, str):
+                    self.value = value if value != '' else None
+                else:
+                    self.value = ParameterInterpolationData.dataToString(value)
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/IdwInterpolation.py]: Would it possibly be nicer to internally keep data and only convert to string when required?

File: python/plugins/processing/tests/AlgorithmsTestBase.py [hunk 4]
```diff
@@ -142,6 +142,14 @@ def load_param(self, param):
                 return [self.load_param(p) for p in param['params']]
             elif param['type'] == 'file':
                 return self.filepath_from_param(param)
+            elif param['type'] == 'interpolation':
+                prefix = processingTestDataPath()
+                tmp = ''
```
Comment (m-kuhn) [inline python/plugins/processing/tests/AlgorithmsTestBase.py]: Could this use the `ParameterInterpolationData` class for this?

Something like

    parameter = ParameterInterpolationData()
    parameter.setValue(param['name'])
    parameter.setPrefixPath(processingTestDataPath())

**Review by m-kuhn — COMMENTED** [review]
Great stuff @alexbruy !
I left some comments to consider, feel free to implement or dismiss them, however you think it's best.

File: python/plugins/processing/algs/qgis/IdwInterpolation.py (hunk 1 above)
Comment (alexbruy) [inline python/plugins/processing/algs/qgis/IdwInterpolation.py]: This is helper function from parameters module, and it uses only inside this module. I will look if we can remove prefix or avoid using it here

**Pull request merged**